Based on:
https://git.openembedded.org/meta-openembedded/commit/meta-multimedia/recipes-connectivity/libupnp/libupnp?id=40622feb2e25a0e1ec72f6fda473d6523c21a7cc

_np variants are not recommended anymore, and glibc 2.34+ will have
these removed

---
 upnp/inc/ithread.h | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/threadutil/inc/ithread.h b/threadutil/inc/ithread.h
index 9e09091..d1dd294 100644
--- a/threadutil/inc/ithread.h
+++ b/threadutil/inc/ithread.h
@@ -55,7 +55,8 @@ extern "C" {
 #endif
 
 
-#if defined(PTHREAD_MUTEX_RECURSIVE) || defined(__DragonFly__)
+#if defined(PTHREAD_MUTEX_RECURSIVE) || defined(__DragonFly__) \
+    || defined (__linux__)
 	/* This system has SuS2-compliant mutex attributes.
 	 * E.g. on Cygwin, where we don't have the old nonportable (NP) symbols
 	 */
@@ -319,7 +320,8 @@ static UPNP_INLINE int ithread_cleanup_thread(void) {
  *      Returns EINVAL if the kind is not supported.
  *      See man page for pthread_mutexattr_setkind_np
  *****************************************************************************/
-#if defined(PTHREAD_MUTEX_RECURSIVE) || defined(__DragonFly__)
+#if defined(PTHREAD_MUTEX_RECURSIVE) || defined(__DragonFly__) \
+    || defined (__linux__)
 	#define ithread_mutexattr_setkind_np pthread_mutexattr_settype
 #else
 	#define ithread_mutexattr_setkind_np pthread_mutexattr_setkind_np

-- 
2.31.1

